flag-gen/templates/cli_flags_generated.gotpl

```diff
@@ -60,6 +60,12 @@ func (f {{ $flag.Name }}Flag) GetValue() string {
 	{{ $flag.ValueString }}
 }
 
+// GetValue returns the flags value as string representation and an empty
+// string if the flag takes no value at all.
+func (f {{ $flag.Name }}Flag) GetTakesFile() bool {
+	{{ if eq $flag.TakesFile true }}return f.TakesFile{{ else }}return false{{ end }}
+}
```

Comment:
Just don't generate these for things that do not take files, and do an interface assertion in the generation code.

Reply:
Hm, the issue is that we have this method as part of the `DocGenerationFlag`, which we already type assert in fish.go. I we remove the function from the other flags we would not be able any more to generate docs/completions for them.

Reply:
Why? You can assert something more than once.

Reply:
Okay, I added a helper function to check if we can access the `TakesFile` variable.